rhel/repositoryscanner_test.go

```diff
@@ -100,6 +122,50 @@ func TestRepositoryScanner(t *testing.T) {
 			cfg:       &RepositoryScannerConfig{API: srv.URL, Repo2CPEMappingURL: srv.URL + "/repository-2-cpe.json"},
 			layerPath: "testdata/layer-with-embedded-cs.tar",
 		},
+		{
+			name: "FromMappingFile",
+			want: []*claircore.Repository{
+				{
+					Name: "cpe:/o:redhat:enterprise_linux:6::server",
+					Key:  repositoryKey,
+					CPE:  cpe.MustUnbind("cpe:/o:redhat:enterprise_linux:6::server"),
+				},
+				{
+					Name: "cpe:/o:redhat:enterprise_linux:7::server",
+					Key:  repositoryKey,
+					CPE:  cpe.MustUnbind("cpe:/o:redhat:enterprise_linux:7::server"),
+				},
+				{
+					Name: "cpe:/o:redhat:enterprise_linux:8::server",
+					Key:  repositoryKey,
+					CPE:  cpe.MustUnbind("cpe:/o:redhat:enterprise_linux:8::server"),
+				},
+			},
+			cfg:       &RepositoryScannerConfig{Repo2CPEMappingFile: f.Name()},
+			layerPath: "testdata/layer-with-embedded-cs.tar",
+		},
+		{
+			name: "FromMappingFileAirGap",
+			want: []*claircore.Repository{
+				{
+					Name: "cpe:/o:redhat:enterprise_linux:6::server",
+					Key:  repositoryKey,
+					CPE:  cpe.MustUnbind("cpe:/o:redhat:enterprise_linux:6::server"),
+				},
+				{
+					Name: "cpe:/o:redhat:enterprise_linux:7::server",
+					Key:  repositoryKey,
+					CPE:  cpe.MustUnbind("cpe:/o:redhat:enterprise_linux:7::server"),
+				},
+				{
+					Name: "cpe:/o:redhat:enterprise_linux:8::server",
+					Key:  repositoryKey,
+					CPE:  cpe.MustUnbind("cpe:/o:redhat:enterprise_linux:8::server"),
+				},
+			},
+			cfg:       &RepositoryScannerConfig{DisableAPI: true, API: esrv.URL, Repo2CPEMappingURL: "/", Repo2CPEMappingFile: f.Name()},
```

Comment:
Can you add a test where we can't get any repo data because there is no file and `DisableAPI: true`?

Reply:
I'm not entirely following, but I tried to add what I thought you were getting at. If the new changes don't seem quite right, can you rephrase your comment?

Reply:
yeah this is what I meant, thanks